### PR TITLE
MNTOR-2742 - Implement Formstack iframe for cancellation survey

### PR DESCRIPTION
**Pull request by codemist**
# References: 
Jira: MNTOR-2742
Figma: https://www.figma.com/file/DTbmXzCQCw2PxXpHQW8yfW/Monitor-MVP-Enhancements?type=design&node-id=424%3A11277&mode=design&t=jCoi4rrjYz2AT06M-1

# Screenshot (if applicable)

https://github.com/mozilla/blurts-server/assets/13066134/fab69f5d-cf2d-4dc6-96bc-75303874529d

# Checklist (Definition of Done)
- [ ] Localization strings (if needed) have been added.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: locales-pending/settings-premium.ftl [hunk 1]
```diff
@@ -22,7 +22,9 @@ settings-cancel-plus-step-survey-note = Please note, all of your { -brand-monito
 settings-cancel-plus-step-survey-cta-label = Continue to cancellation
 settings-cancel-plus-step-survey-cancel-label = Never mind, take me back
 settings-cancel-plus-step-redirecting-heading = We appreciate your feedback
-settings-cancel-plus-step-redirecting-lead = We’ll automatically redirect you to your { -brand-mozilla-account } where you can cancel your { -brand-monitor-plus } subscription.
+settings-unsubscribe-dialog-confirmation-redirect-title = Directing you to your { -brand-mozilla } account to cancel
+settings-unsubscribe-dialog-confirmation-redirect-description-pt1 = We’ll automatically redirect you to your { -brand-mozilla } account where you can cancel your { -brand-monitor } subscription.
```
Comment (pdehaan) [inline locales-pending/settings-premium.ftl:26]: Q: should this be `{ -brand-mozilla} account` or `{ -brand-mozilla-account }`?

```sh
git grep -Pin "^-brand-mozilla(-account)?\s" locales/en locales-pending/ | cat

locales/en/app.ftl:14:-brand-Mozilla = Mozilla
locales/en/app.ftl:865:-brand-mozilla = Mozilla
locales/en/app.ftl:879:-brand-mozilla-account = Mozilla account
```

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/CancelFlow.module.scss [hunk 2]
```diff
@@ -11,13 +11,32 @@
...
+    height: 100%;
+  }
 }
 
-.surveyIframe {
+.iframeWrapper {
   width: 100%;
-  height: 350px;
+  overflow: hidden;
+  transform: rotate(0.00001deg);
```
Comment (pdehaan) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/CancelFlow.module.scss:38]: Interesting, what does the _almost_-rotation do?

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/CancelFlow.tsx [hunk 3]
```diff
@@ -132,48 +125,46 @@ export const CancelFlow = (props: Props) => {
...
-                    {l10n.getFragment("settings-cancel-plus-step-survey-note", {
-                      elems: { b: <b /> },
-                    })}
-                  </p>
-                  {/*
-                  // Replaced by listening to survey-complete events:
+
+                  <div className={styles.iframeWrapper}>
+                    <iframe
+                      scrolling={"no"}
```
Comment (pdehaan) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/CancelFlow.tsx:131]: Any difference between `{"no"}` and `"no"`?

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/CancelFlow.tsx [hunk 4]
```diff
@@ -132,48 +125,46 @@ export const CancelFlow = (props: Props) => {
...
+                      frameBorder={0}
+                      src="https://mozilla.formstack.com/forms/mozilla_monitor_plus_cancel"
+                      title="Formstack Form"
```
Comment (pdehaan) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/CancelFlow.tsx:134]: Should this be localized?

File: locales-pending/settings-premium.ftl [hunk 5]
```diff
@@ -22,7 +22,9 @@ settings-cancel-plus-step-survey-note = Please note, all of your { -brand-monito
 settings-cancel-plus-step-survey-cta-label = Continue to cancellation
 settings-cancel-plus-step-survey-cancel-label = Never mind, take me back
 settings-cancel-plus-step-redirecting-heading = We appreciate your feedback
-settings-cancel-plus-step-redirecting-lead = We’ll automatically redirect you to your { -brand-mozilla-account } where you can cancel your { -brand-monitor-plus } subscription.
+settings-unsubscribe-dialog-confirmation-redirect-title = Directing you to your { -brand-mozilla } account to cancel
```
Comment (flodolo) [inline locales-pending/settings-premium.ftl:25]:
```suggestion
settings-unsubscribe-dialog-confirmation-redirect-title = Directing you to your { -brand-mozilla-account } to cancel
```

All the instances of `{ -brand-mozilla } account`  are mistakes that should be fixed over time.

File: locales-pending/settings-premium.ftl (hunk 1 above)
Comment (flodolo) [inline locales-pending/settings-premium.ftl:26]:
```suggestion
settings-unsubscribe-dialog-confirmation-redirect-description-pt1 = We’ll automatically redirect you to your { -brand-mozilla-account } where you can cancel your { -brand-monitor } subscription.
```

File: locales/en/settings.ftl [hunk 6]
```diff
@@ -84,6 +84,6 @@ settings-unsubscribe-dialog-title = We’re sorry to see you go. <br /> Will you
 settings-unsubscribe-dialog-info = Your experience is important to us. We read every response and take it into consideration.
 settings-unsubscribe-dialog-message-placeholder = What could have gone better?
 # $faq_href is the URL for the faq page. HTML tags should not be translated, e.g. `<a>`
-settings-unsubscribe-dialog-confirmation = Please note, all of your { -brand-monitor-premium } services will be <a { $faq_href }>permanently deleted</a> after your current billing cycle ends.
+settings-unsubscribe-dialog-confirmation = Please note, all of your { -brand-monitor-plus } services will be <a { $faq_href }>permanently deleted</a> after your current billing cycle ends.
```
Comment (flodolo) [inline locales/en/settings.ftl:87]: You need a new string ID to make this type of change.
https://mozilla-l10n.github.io/documentation/localization/making_string_changes.html

**Review by flodolo — CHANGES_REQUESTED** [review]

**Review by Vinnl — COMMENTED** [review]
Added some inline comments, but I'd like to see it with `main` as the target, just to make sure none of the cruft I added while spiking is still in there if it's no longer relevant :)

But: it looks great in the video! Much, much better than what I had before.

image

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/SettingsPage.test.tsx [hunk 7]
```diff
@@ -449,7 +449,7 @@ it("shows the Plus cancellation link if the user has Plus", () => {
   expect(cancellationHeading).toBeInTheDocument();
 });
 
-it("takes you through the cancellation dialog flow", async () => {
+it("takes you through the cancellation dialog flow all the way to sublat", async () => {
```
Comment (pdehaan) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/SettingsPage.test.tsx:452]:
```suggestion
it("takes you through the cancellation dialog flow all the way to subplat", async () => {
```

**Review by flodolo — APPROVED** [review]

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/View.tsx [hunk 8]
```diff
@@ -128,6 +133,7 @@ export const SettingsView = (props: Props) => {
                   link_id: "deactivate_account",
                 }}
               >
+                {/* changed string here */}
```
Comment (Vinnl) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/View.tsx:136]:
```suggestion
```

**Review by Vinnl — APPROVED** [review]
The top of the form gets cut off on small screens, but I'm not sure if that's a blocker.

Screen Shot 2024-04-17 at 21 49 08

The unit tests look really good!

**Pull request merged**